### PR TITLE
Fixing static analyzer warnings in ElectronEnergyCorrector.cc, GEDPhotonProducer.cc and PhotonMIPHaloTagger.cc

### PR DESCRIPTION
**Pull request by uhussain**
In all three files, dead assignment errors raised by static analyzer are fixed by deleting these assignments. These preliminary fixes pass all tests, however some deliberation is needed for the file PhotonMIPHaloTagger.cc which has other similar "dead assignments" which are not pointed out by the static analyzer. Or perhaps, I misunderstand the intention of the code. @slava77 and @cvuosalo you requested to watch this.

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoEgamma/PhotonIdentification/src/PhotonMIPHaloTagger.cc [hunk 1]
```diff
@@ -290,17 +290,11 @@ std::vector<double> PhotonMIPHaloTagger::GetMipTrailFit(const reco::Photon* phot
            
 
         //Lets Initialize them first for each iteration 
-        res    = 0.0;
-        res_sq = 0.0;
-        wt     = 0.0;
```
Comment (VinInn) [inline RecoEgamma/PhotonIdentification/src/PhotonMIPHaloTagger.cc]: Sincerely I thing we are going well beyond what to expect form the average developers and what we SHALL expect from the compiler's optimizer:
Assigning zero to a variable does not harm (and in many other contest we ask developers to initialize variables)
And the compiler is free to optimize it away if its dependency analysis discover that it is redundant...

Comment (uhussain) [inline RecoEgamma/PhotonIdentification/src/PhotonMIPHaloTagger.cc]: @VinInn that is indeed a fair point since there are other variables that are initialized similarly and the static analyzer gives no warnings about them.

Comment (VinInn) [inline RecoEgamma/PhotonIdentification/src/PhotonMIPHaloTagger.cc]: of course it is useful to identify and remove any costly dead assignment ```log(sqrt(exp(sin(myfunc(nonCont/anotherNonConst))))```
(still a good compiler should optimixe it away. Unfortunately myfunc may have side effects
so it will not (and I am not sure the static analyser will report it...)

**Comment by davidlange6** [comment]
I agree - looking more closely at the changes, I think the warnings are a sign of code that can be improved, but not in the way done in this PR. These are potentially the "right" fixes if trying to get rid of warnings from this tool

ElectronEnergyCorrector.cc: declare newEnergy where it is used (around line 101)
PhotonMIPHaloTagger.cc
declare res where it is used around line 324
make res_sq a local variable to the loop around line 325
make wt a local variable to the loop around line 325

basically to give variables the scope they need.

**Comment by davidlange6** [comment]
-1 in its current state. initializing variables is good practice

**Comment by cvuosalo** [comment]
@uhussain: For the six variables flagged in PhotonMIPHaloTagger.cc, please move the definition of each variable to where it is first set with a useful value. That limits the scope of each variable to where it is actually used, and the zero initializations will no longer be needed.

**Comment by cvuosalo** [comment]
@cmsbuild please test

**Comment by davidlange6** [comment]
+1
